gltf-derive/src/lib.rs

```diff
@@ -4,31 +4,31 @@
 
 #![recursion_limit = "128"]
 
-#[macro_use]
-extern crate quote;
 extern crate proc_macro;
 
 use proc_macro::TokenStream;
+use syn::DeriveInput;
 
 #[proc_macro_derive(Validate)]
 pub fn derive_validate(input: TokenStream) -> TokenStream {
-    let source = input.to_string();
-    let ast = syn::parse_macro_input(&source).unwrap();
-    let tokens = expand(&ast);
-    tokens.parse().unwrap()
+    expand(&syn::parse_macro_input!(input as DeriveInput)).into()
 }
 
-fn expand(ast: &syn::MacroInput) -> quote::Tokens {
-    let fields = match ast.body {
-        syn::Body::Struct(syn::VariantData::Struct(ref fields)) => fields,
+fn expand(ast: &DeriveInput) -> proc_macro2::TokenStream {
+    use proc_macro2::TokenStream;
+    use quote::quote;
+
+    let fields = match ast.data {
+        syn::Data::Struct(ref data_struct) => &data_struct.fields,
         _ => panic!("#[derive(Validate)] only works on `struct`s"),
     };
     let ident = &ast.ident;
-    let minimal_validations: Vec<quote::Tokens> = fields.iter()
+    let minimal_validations: Vec<TokenStream> = fields
+        .iter()
         .map(|f| f.ident.as_ref().unwrap())
         .map(|ident| {
             use inflections::Inflect;
-            let field = ident.as_ref().to_camel_case();
+            let field = ident.to_string().to_camel_case();
```

Comment:
Why is `to_string` needed?

Reply:
Using .as_ref() to get a `str` was [removed in syn 0.13](https://github.com/dtolnay/syn/commit/a74a1c89fd76b4f2ba09f6dddf916287b0565a50#diff-2e7b2e9a81356f6d913575ff00a00e87), because [proc_macro2 0.4 removed](https://github.com/alexcrichton/proc-macro2/commit/f388843b8a92b5523c30d51f1da436a05ebb5551#diff-b4aea3e418ccdb71239b96952d9cddb6) their `Ident` `AsStr` impl, as the normal proc_macro crate removed their `AsStr` impl, the rationale for which is explained [here](https://github.com/rust-lang/rust/issues/38356#issuecomment-386786645).